File: solutions/typescript/pangram/1/pangram.ts
class Pangram {
    constructor(readonly input: string) {}

    isPangram = () => {
      const lowercaseAlphabet = "abcdefghijklmnopqrstuvwxyz";
      return [...new Set([...this.input.toLowerCase().replace(/\s|_|\d|\.|\"|[^a-z]/g, "")])].sort().join('') === lowercaseAlphabet;
    }
  }
  
export default Pangram

File: solutions/kotlin/pangram/1/src/main/kotlin/Pangram.kt
class Pangram {
    companion object {
        fun isPangram(x: String): Boolean {
            if (x.length < 26) return false
            val t = x.toLowerCase()
            for (c in 'a' .. 'z')
                if (c !in t) return false
            return true
        }
    }
}